
<img src="https://coursesandconferences.wellcomeconnectingscience.org/wp-content/themes/wcc_courses_and_conferences/dist/assets/svg/logo.svg" width="300" height="50"> 

# Genomics and Epidemiological Surveillance of Bacterial Pathogens - Latin America and the Caribbean

Learn how to apply genomics to epidemiological surveillance of bacterial pathogens and public health in Latin America

## 16–21 April 2023, Asunción, Paraguay

[Genomics and Epidemiological Surveillance of Bacterial Pathogens Course Website Link](https://coursesandconferences.wellcomeconnectingscience.org/event/genomics-epidemiological-surveillance-of-bacterial-pathogens-latin-america-and-the-caribbean-20230416/) <br /> 
[Course Timetable](https://github.com/WCSCourses/GenEpiLAC2023/blob/main/Manuals/Timetable.pdf) <br /> 
Link to [Course Manual](https://github.com/WCSCourses/GenEpiLAC2023/blob/main/Manuals/README.md)

# Summary

In collaboration with the [Central Laboratory of Public Health](https://www.mspbs.gov.py/lcsp), Ministry of Health and the Faculty of Exact and Natural Sciences of the [National University of Asunción](https://www.una.py/english), Paraguay, we are pleased to announce our latest overseas course in Genomics and Epidemiological Surveillance of Bacterial Pathogens.

This course aims to provide training in the skills required to generate and interpret high throughput genome sequencing data in a public health setting. This year’s course will have a strong emphasis on AMR and pathogens endemic to Latin America, whilst also training participants in the latest laboratory and computational methods for genomic sequence analysis and surveillance.

# Course background

This is a focused course that was established in 2013 to show how traditional epidemiological surveillance, molecular typing and genomics based methods complement each other and hence contribute to generating the most reliable information available for public health decision making. Although the course is built around genomic surveillance the focus of this course is designed to reflect regional public health priorities on bacterial pathogens, and this year’s course will have a strong emphasis on infections endemic to Latin America. In the last few years the surveillance community faced the challenge of the COVID-19 pandemic, speeding the process of introducing advances in genomics, sequencing technology and our ability to interpret high resolution sequence data for surveillance purposes.

We are now seeing a shift from the use of molecular typing data for epidemiological surveillance and public health investigations of infectious disease to using Whole Genome Sequencing (WGS) for this purpose. In some settings WGS is now being used to define outbreaks and trace infection sources in humans, animals and the environment, contributing to reinforce a One Health approach. This approach also offers enormous promise for predicting antimicrobial resistance phenotypes.

Existing studies that have used genomics for mapping the spread of pathogens have highlighted two things, firstly that genomic datasets are growing day by day and they offer the context against which to understand local patterns of disease. Secondly these studies also highlight the power gained from public health scientists and academic research scientists working together to better understand infectious disease. This is the foundation of this course and expert instructors have been assembled from a wide range of settings in Latin America and Europe, including national and international public health organisations and front line high-profile genomic research organisations. We recognise the synergy between academic research and public health and this is your opportunity to strengthen this effort by attending the course.

# Target audience

The course is targeted at clinical/public health molecular biologists and microbiologists working in Latin America and the Caribbean.

The course is free to attend for non-commercial applicants. Limited travel bursaries are also available.

# Programme

This course builds upon previous versions, incorporating genomics and problem solving exercises with modules that demonstrate how epidemiological information and phenotyping-in combination with sequence data – can be used:

1. for georeferencing (also known as ‘phylogeography’) where the location of disease causing strains are mapped with detailed genetic and phenotypic data (for example antibiotic resistance patterns) to look for regional patterns of disease, and
2. to understand the fine detail of new and emerging infections, the risk to public health and the management of infectious diseases.

**The week-long programme will include:**

- Practical bioinformatics skills needed to manipulate raw sequence data (some of which will be generated on the course), including: mapping and de novo assemblies, SNP calling, annotation, antimicrobial resistance calling, and analysis of genomic information
- Principles of phylogeny and construction of accurate phylogenies
- Group exercises designed to teach everyday skills required to work in genomic surveillance
- Practical exercises for using GPS to map cases of infectious disease and phenotypes (such as AMR) across the world

**Learning outcomes**

After attending the course, participants should be able to:

- Recognise the principles of high throughput sequencing and how to apply them effectively
- Sequence genomic bacterial DNA using short- and long-read methods
- Use sequence analysis tools effectively
- Explain what a phylogenetic tree is, and what it represents
- Formulate projects using the global surveillance platform Microreact
- Perform comparative genomics on multiple genome sequences
- Compare different traditional bacteriological identification methods with molecular typing methods and genomics in the context of local and regional resources
- Explain, using examples, how bacterial pathogens and AMR spread among human populations.
 

*Please note: The practical sessions will be taught exclusively through Unix/Linux. Therefore, participants are required to have some familiarity with the Linux operating system. This will be essential for participants to fully benefit from the course. There are numerous online introductory tutorials to the UNIX/Linux operating system and command line, including:*

[http://www.ee.surrey.ac.uk/Teaching/Unix](http://www.ee.surrey.ac.uk/Teaching/Unix)

[http://swcarpentry.github.io/shell-novice/](http://swcarpentry.github.io/shell-novice/)

**Course Instructors**      

- [Mathew Beale](https://www.sanger.ac.uk/person/beale-mathew/), Wellcome Sanger Institute, UK
- [Josefina Campos](https://www.researchgate.net/profile/Josefina-Campos-2), CNGB ANLIS “Dr. Carlos G. Malbrán, Argentina
- [Julio Diaz Caballero](https://www.pathogensurveillance.net/team/), Centre for Genomic Pathogen Surveillance, University of Oxford, UK
- [Matthew Dorman](https://www.sanger.ac.uk/person/dorman-matthew/), Wellcome Sanger Institute, UK
- [Caterina Guzmán-Verri](https://scholar.google.com/citations?user=S1GNAC8AAAAJ&hl=en), PIET, Veterinary School, Universidad Nacional, Costa Rica
- [Matt Holden](https://risweb.st-andrews.ac.uk/portal/en/persons/matthew-holden(029e1abe-7829-4bc3-accf-9d9e392f274c).html), University of St Andrews, Scotland
- [Tomas Poklepovich](https://www.researchgate.net/profile/Tomas-Poklepovich), CNGB ANLIS “Dr. Carlos G. Malbrán, Argentina
- [Maria Sol Haim](https://www.researchgate.net/profile/Maria-Sol-Haim), CNGB ANLIS “Dr. Carlos G. Malbrán, Argentina
- [Marcela Suárez-Esquivel](https://www.researchgate.net/profile/Marcela-Suarez-Esquivel), Veterinary School, Universidad Nacional, Costa Rica
- Natalie Weiler Gustafson, Central Laboratory of Public Health Asunción, Paraguay


******

## Citing and Re-using Course Material

The course data are free to reuse and adapt with appropriate attribution. All course data in these repositories are licensed under the <a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0)</a>. <a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br /> 

## Interested in attending a course?

Take a look what courses are coming up at [Wellcome Connecting Science Courses & Conference Website](https://coursesandconferences.wellcomeconnectingscience.org/our-events/).

---

For more information or queries, feel free to contact us via the [Wellcome Connecting Science website](https://coursesandconferences.wellcomeconnectingscience.org).<br /> 


Find us on socials [Wellcome Connecting Science Linktr](https://linktr.ee/eventswcs)

---

[Wellcome Connecting Science GitHub Home Page](https://github.com/WCSCourses) <br /> 
