<img src="https://coursesandconferences.wellcomeconnectingscience.org/wp-content/themes/wcc_courses_and_conferences/dist/assets/svg/logo.svg" width="200" height="200">


# Web Tools for Genomic Epidemiology - Paraguay 2023 <!-- omit in toc -->

### Module Lead: Silvia Argimon & Julio Diaz Caballero <!-- omit in toc -->
<br>

# Table of contents <!-- omit in toc -->
- [Background](#background)
- [Introduction](#introduction)
- [Learning outcomes](#learning-outcomes)
  - [Collect field data with Epicollect](#collect-field-data-with-epicollect)
  - [Merge data with Data-flo](#merge-data-with-data-flo)
  - [Microreact](#microreact)
- [Resources](#resources)
  - [Install and test Epicollect5 on your mobile phone](#install-and-test-epicollect5-on-your-mobile-phone)
  - [Sign-in to data-flo](#sign-in-to-data-flo)
  - [Sign-in to Microreact](#sign-in-to-microreact)
  - [Web colours](#web-colours)
- [Bonus activities](#bonus-activities)

# Background
Genomic epidemiology aims to understand the emergence and dissemination of high-risk clones within pathogen populations with the ultimate goal of implementing evidence-based interventions to protect public health. High-risk clones are subpopulations/strains/variants of a pathogen that carry risk elements, such as antibiotic resistance or virulence determinants, and thus pose a potential risk to public health.

The increased genetic resolution afforded by genomic data is useful from global to local geographic scales, and has proven to be particularly useful for the investigation of pathogens that exhibit little genetic variation (e.g. [Wong *et al.* 2016](https://pubmed.ncbi.nlm.nih.gov/27703135/)), and for outbreak investigations (e.g. [Hendriksen *et al.* 2011](https://pubmed.ncbi.nlm.nih.gov/21862630/), [Eppinger *et al.* 2014](https://pubmed.ncbi.nlm.nih.gov/25370488/)). During outbreak investigations (and other genomic epidemiology studies), epidemiological data from patients is collected by healthcare professionals. Antimicrobial susceptibility data, species identification, and any further phenotypic or molecular characterization of the isolates is often generated by the laboratories linked to healthcare facilities and/or by the reference laboratory. Ideally, these different sources of data are stored in a centralised surveillance system and database, such as WHONET. However, these systems rarely incorporate genomic data produced by bioinformaticians. Genomic data may include sequence/assembly quality, genotyping/genoserotyping information, presence/absence of known risk elements, and clustering of the isolates based on genetic similarity –usually in the form of a phylogenetic tree or a minimum spanning-tree.

The job of a genomic epidemiologist often starts by combining data from diverse sources that might not be complete or standardised to facilitate the identification of relevant patterns for meaningful interpretation. Once integrated, the epi, lab, and genomic data can be interrogated over the structure of the tree to identify clusters of interest that will place the isolates in or out of the outbreak.

The **[Centre for Genomic Pathogen Surveillance](https://www.pathogensurveillance.net/our-software/)** develops free web applications for data collection, integration, visualisation, and analysis of genomic epidemiology data. **[Epicollect5](https://five.epicollect.net/)** is a mobile & web application for free and easy data collection. It provides both the web and mobile applications for the generation of forms (questionnaires) and freely hosted project websites for data collection. Projects are created by using the web application at five.epicollect.net, and then downloaded to the device to perform the data collection. Data are collected in the field using multiple devices and all data can be viewed on a central server (via map, tables, and charts). **[Data-flo](https://data-flo.io/)** is a system for customised integration and manipulation of diverse data via a simple drag and drop interface. Data-flo provides a visual method to design a reusable pipeline to integrate, clean, and manipulate data in a multitude of ways, eliminating the need for continuous manual intervention (e.g., coding, formatting, spreadsheet formulas, manual copy-pasting). **[Microreact](https://microreact.org/)** allows you to upload, visualise and explore any combination of clustering (trees), geographic (map) and temporal (timeline) data. Other metadata variables are displayed in a table. You can specify colours and/or shapes to display on the map, tree and/or timeline. A permanent URL is produced for you to share your Microreact.

# Introduction
Aim:  To use web and mobile tools for the collection of epi data in the field, its integration with lab and bioinformatics data, and to finally visualise it together to detect a putative outbreak.

*Note: This exercise is a work of fiction. Any resemblance to a real life situation is purely coincidental.*

During January and February 2023, fifty-eight patients reported to eight hospitals in Asunción with acute watery diarrhoea and vomiting. *Shigella sonnei* was isolated from the stool samples of fourty patients (all belong to CTX-M-27 producing *S. sonnei* ST152), and one colony from each confirmed case was sequenced on Illumina MiSeq. An epidemiological investigation revealed that more than half of the culture-confirmed cases regularly visited the Ñu Guazú Park, where water fountains and other water sources are accessible, and which is crossed by the Arroyo Itay. This suggested that these infections could be related to contaminated water in the park. The epidemiological and demographic data is summarised on the file epi_data.csv.

A team of disease detectives will be deployed around Ñu Guazú Park today to investigate the potential source of these infections using Epicollect5, data-flo, and Microreact, free mobile and web applications developed by the Centre for Genomic Pathogen Surveillance for easy data collection, integration and visualisation.

# Learning Outcomes
At the end of this session the participants will be able to:

1. Collect data in the field with Epicollect5.
2. Merge epidemiological, laboratory and genomic data with data-flo
3. Create a Microreact project to visualise epidemiological, laboratory and genomic data

## 1. Collect field data with Epicollect
`https://five.epicollect.net/`

The disease detectives will work in pairs, and will be provided with a **collection tube** with a **barcode** sample identifier to collect a sample of water, and will use their **mobile phone** to collect associated metadata in the field.

Make sure your phone is connected to WiFi.
Open the **Epicollect5** app on your mobile phone.

![](epicollect_logo.jpeg)

<br>

Add the project **Asuncion 2023 S sonnei Outbreak** by typing into the search box --you will need to be connected to WiFi.

![](epicollect_search.jpeg)

<br>

Using the map of the Ñu Guasú park provided, walk to the location of your collection point (water source). Locations are indicated by the black dots labelled a-j. You will not need internet access to collect data in the field.

<img src="location_map.png" width=50% height=50%>

<br>

Open the **Asuncion 2023 S sonnei Outbreak** project and add an entry. Follow the form to collect a water sample and associated metadata.

<img src="epicollect_add_entry.jpg" width=40% height=40%>


<br>

Your team will collect only one water sample, but both of you can collect the associated data on Epicollect5. However, *make sure that only one of you uploads the data later on to avoid duplications.*

Return to the Epidemic Intelligence Center (i.e. the classroom).

One of the pair members only. **Upload your entry (and image)** to the Epicollect5 server --make sure you are connected to the WiFi.

<img src="epicollect_upload_data.jpeg" width=30% height=30%>

<br>

Submit your water sample to the instructor, which will be sent to the reference lab for culture confirmation.

Once all entries are uploaded by the disease detectives we will take a look at the data together on **[https://five.epicollect.net/project/asuncion-2023-s-sonnei-outbreak](https://five.epicollect.net/project/asuncion-2023-s-sonnei-outbreak)**

Answer the following questions:

> 1. What is the most common type of water source?
> 2. Was it possible to collect a sample from all 13 sources? If not, what were the reasons?


## 2. Merge data with Data-flo
`https://data-flo.io/`

*Note: you need to sign-up for data-flo and Microreact. See instructions in the [Resources](#resources) section. Creating your own account will allow you to manage and edit your projects.*

The reference lab has sent you the culture results on an Excel file called ([`lab_results.xlsx`](https://drive.google.com/drive/folders/1KMbks4oWsFRlB2Qt48MkZnTMLJTUB13f?usp=share_link)). The lab reported that **2 out of 13** water samples from Ñu Guasú park were positive for *S. sonnei*. **This immediately prompted the closure of the 2 water sources.**

> Does this confirm that the source of the outbreak was contaminated water from Ñu Guasú park?

One colony from each source was sequenced on Illumina MiSeq by the reference lab. A maximum likelihood phylogenetic tree ([`tree.nwk`](https://drive.google.com/drive/folders/1KMbks4oWsFRlB2Qt48MkZnTMLJTUB13f?usp=share_link)) was inferred from the genomes of the 34 clinical samples and 2 environmental (water) samples. Six genomes from a previous outbreak (Jan-Feb 2020) were also included in the tree inference and their associated data added to the ([`epi_data.csv`](https://drive.google.com/drive/folders/1KMbks4oWsFRlB2Qt48MkZnTMLJTUB13f?usp=share_link)) file.

The disease detectives now have the information needed for the investigation in the following formats:

*  `epi_data.csv` Epi data from 34 clinical cases and 6 cases from previous outbreak
*  `Epicollect project` Metadata of 13 water sources from Ñu Guasú Park
*  `lab_results.xlsx` Culture and serotyping results
*  `tree.nwk` Phylogenetic tree of 34 clinical cases, 2 culture-positive water samples, and 6 cases from previous outbreak

The files are located in **[this google drive link](https://drive.google.com/drive/folders/1KMbks4oWsFRlB2Qt48MkZnTMLJTUB13f?usp=share_link)**.

We will combine data from these different sources with a data-flo workflow that takes the files above and the data from the Epicollect project as input, and creates as an output a Microreact project where the data can be visualised.

Open the data-flo workflow (**[https://data-flo.io/run?prnvpvmKzq8rDKbPD9ZXAs](https://data-flo.io/run?prnvpvmKzq8rDKbPD9ZXAs)**)

<img src="dataflo_2023.png" width=50% height=50%>

<br>

Copy the workflow to your own dataflo account.

<img src="dataflo_copy.png" width=50% height=50%>

<br>

This will open a copy of this workflow in your dataflo account.

<img src="dataflo_workflow.PNG" width=50% height=50%>

<br>

**On a different browser tab**, get your microreact `API access token` at **[https://microreact.org/my-account/settings](https://microreact.org/my-account/settings)** (you must already have created your microreact account).

<img src="miroreact_token.png" width=50% height=50%>

<br>

Edit the workflow to include your own microreact `API access token`.

1. Click on `*access token` in the `Create microreact project` box
2. From the options on the right, select `Bind to value`
3. Paste your `API access token` in the `VALUE` box

<img src="dataflo_key.png" width=50% height=50%>

<br>

Save your newly edited dataflo project by clicking on the `save` icon.

<img src="dataflo_save.png" width=50% height=50%>

<br>

Now lets go to the implementation page by clicking on the `i` icon.

<img src="dataflo_implementation.png" width=50% height=50%>

<br>

To run the workflow paste the url of the Epicollect5 project (**[https://five.epicollect.net/project/asuncion-2023-s-sonnei-outbreak](https://five.epicollect.net/project/asuncion-2023-s-sonnei-outbreak)**) and upload the files, found [`here`](https://drive.google.com/drive/folders/1KMbks4oWsFRlB2Qt48MkZnTMLJTUB13f?usp=share_link).

<img src="dataflo_run2.png" width=50% height=50%>

<br>

Click on **`Run`**. The Outputs box now shows the url of a Microreact project created by data-flo.

<img src="dataflo_result.png" width=50% height=50%>

Before you open the Microreact project answer the following questions:

> 1. If later on we wanted to add additional entries (i.e. water samples) to the Epicollect project, could we use the same data-flo? If so, how would this be beneficial?
> 2. Do you use/know of any other methods to join data? If so, how do they compare with data-flo?

## 3. Microreact
`https://microreact.org/`

Open the Microreact link in the output from data-flo. It should look something like this:

<img src="microreact_link.png" width=50% height=50%>

<br>

Now change the colour column to “Hospital”. Click on the eye icon at the top left of the page. Select "Hospital from the `Colour Column` dropdown list.

<img src="microreact_colour.png" width=50% height=50%>


<br>

Scale the markers on the map by clicking on the sliders icoc, then in the opening options click on `Markers`. Finally, toggle the `Scale markers` option on.

<img src="microreact_markers.png" width=50% height=50%>


<br>

Answer the following questions:

> 1. What do the markers on the map represent?
> 2. Which hospital reported the index (first) case? Tip: Click on the earliest sample on the timeline.

Change the colour column to “Source”. Click on the eye icon at the top left of the page. Then select "Source" from the `Colour Column` dropdown list.

<img src="microreact_source.png" width=50% height=50%>

<br>

> 3. Can you confirm that the source of the outbreak is the contaminated water from Ñu Guasú Park?
> 4. Why is it useful to include genomes from a previous outbreak in the analysis? Is the current outbreak related to the past outbreak from Jan-Feb 2020?

Microreact assigns colours automatically to every column in your metadata table, but you can customise them. Let's change the colour column to “Household”. Click on the eye icon at the top left of the page. Then select "Household" from the `Colour Column` dropdown list.

<img src="microreact_household.png" width=50% height=50%>

<br>

Now Select "Categorical" from the `Colour Palette` option. Next Select "qualitative" from the `Palette type` dropdown list and "10" from the `Number of colours` dropdown list. Finally, select the second option from the list of palettes.

<img src="microreact_palette.png" width=50% height=50%>

<br>

Change the colour column back to “Hospital” and add the columns “Source” and “Household” as metadata blocks. For this, select the slider icon from top right of the map panel, then, click on the `Metadata blocks` button. Finally, select "Source" and "Household" from the list of options.  

<img src="microreact_metadata_blocks.png" width=50% height=50%>

<br>

> 5. Do any of the cases share a household? Taking into account the incubation period for S. sonnei (a few hours to five days), does the data support intra-household transmission?


# Resources

## Install and test Epicollect5 on your mobile phone
Epicollect5 is available on Android and iOS. To install it on your mobile device, get it from Google Play or the Apple App Store.
To test Epicollect5 on your phone, open the application and click on the pre-loaded EC5 Demo Project. Click on `+ Add entry` and follow the questionnaire. Please answer all the questions, including taking a photo with your phone’s camera. Once you’ve answered the questions, save and upload the entry. Please note that you must be connected to the internet (or to a mobile network) to be able to upload an entry.

## Sign-in to data-flo
**[https://docs.data-flo.io/introduction/getting-started-sign-in](https://docs.data-flo.io/introduction/getting-started-sign-in)**

## Sign-in to Microreact
Head over to **[https://microreact.org/my-account](https://microreact.org/my-account)**, and sign up using your email address or any other authenticating option

!<img src="microreact_signup.png" width=30% height=30%>

## Web colours
Web colours are used on web pages, such as the microreact.org projects, and are usually specified in hexadecimal format preceded by ​​a number sign (or hashtag sign if you were born before the Jurassic period). For example, `#FFFFFF` is the hexadecimal code (or hex code) for the colour “white”, `#000000` is the hex code for the colour “black”, and  `#FF00FF` is the hex code for the colour “magenta”. For more information on the format of the hexadecimal code see Wikipedia page **[https://en.wikipedia.org/wiki/Web_colors](https://en.wikipedia.org/wiki/Web_colors)**

Useful websites to work with web colours:

### ColorBrewer
**[https://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3](https://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3)**
Colour palettes for maps

### ColorHexa
**[https://www.colorhexa.com/](https://www.colorhexa.com/)**
Get complementary colours, colour gradients, etc.

### Coolors
**[https://coolors.co/palettes/trending](https://coolors.co/palettes/trending)**
Colour palette generator

### Wes Anderson colour palettes
**[https://github.com/karthik/wesanderson](https://github.com/karthik/wesanderson)**
Hipster colour palettes inspired by Wes Anderson movies


## Bonus activities

### 1. Download data from the Epicollect5 project on the web.

a. Go to the **[epicollect link provided above](https://five.epicollect.net/project/asuncion-2023-s-sonnei-outbreak)**.

b. Select `VIEW DATA`.

<img src="epicollect_view_data.PNG" width=50% height=50%>

c. Click on `Download` and select your format of choice.

 <img src="epicollect_download.png" width=50% height=50%>

### 2. Create a microreact project using a metadata in csv format and a phylogenetic tree.

a. Click on the menu icon at the top left of the screen

 <img src="microreact_menu_icon.png" width=40% height=40%>

b. Click on `Upload`

 <img src="microreact_upload_option.png" width=30% height=30%>

c. Click on the `+` symbol at the bottom right of the screen.

 <img src="microreact_plus_icon.png" width=10% height=10%>

d. Select `Browse Files`

 <img src="microreact_browse_files.png" width=20% height=20%>

e. Choose the files `tree.nwk` and `epi_data.csv`, and click on the `Open` button (These files were downloaded in the [dataflo](#merge-data-with-data-flo) section).

f. Click on `Continue` in the next prompts.

g. Your view will look like this:

<img src="microreact_bonus_view.png" width=30% height=30%>

h. You can add a Timeline by clicking on the `pencil` icon at the top left of the screen, and clicking on `Create New Timeline`

<img src="microreact_pencil_icon.png" width=50% height=50%>

i. Then position the new panel by dragging the pointer to the bottom panel.

<img src="microreact_bottom_panel.png" width=50% height=50%>

j. Finally, select "One column: Formatted Values" from the `Temporal Data Type` dropdown list, and "Collection Date" from the `Temporal Data Column` dropdown list. Click on `CLOSE`

<img src="microreact_temp_data_type.png" width=50% height=50%>


## 3. Create a microreact project from a Google Spreadsheet

a. Open this **[Google spreadsheet](https://docs.google.com/spreadsheets/d/1DN6fyhHmmkgEU0Ml7Ep_y_2l_JpQZkBfmXZzEwRs70Y/edit?usp=sharing)**

<img src="bonus_original_spreadsheet.png" width=50% height=50%>

b. Make a copy of this in your own Google account by selecting `Make a copy` from the `File` menu item. When prompted, click on `Make a copy`. This will open a separate tab where the copy will be available.

<img src="bonus_make_copy.png" width=30% height=30%>

<img src="bonus_copy.png" width=30% height=30%>

c. Set access to shareable by clicking on the `Share` button at the top right of the screen. A dialogue screen will popoup, here, click on the `Restricted` button, select `Anyone with the link`, and click on `Done`

<img src="bonus_share.png" width=30% height=30%>

d. Now to publish the google spreadsheet click on the `File` menu item, select `Share`, and click on `Publish on web`

<img src="bonus_publish.png" width=30% height=30%>

e. In the popup message click on `Web page`, and select `Comma-separated values (.csv)`. Also, make sure the `Automatically republish when changes are made` option is turned on under the `Published content and settings` section

<img src="bonus_csv.png" width=30% height=30%>

<img src="bonus_auto_publish.png" width=30% height=30%>

f. Confirm your choices in the popup message clicking on `OK`. This will provide more details about your Google spreadsheet, copy the url.

<img src="bonus_publish_sure.png" width=30% height=30%>

<img src="bonus_link.png" width=30% height=30%>

g. Go to **[microreact.org](https://microreact.org)**, and select `Upload` from the main menu.

<img src="bonus_upload.png" width=10% height=10%>

h. Select the plus icon at the bottom right, and click on `Add URLs`

<img src="bonus_add_url.png" width=30% height=30%>

i. Paste the url from step *f* and select `Data (CSV or TSV)` under `File kind`. Finally click on `CONTINUE` to see your microreact.

<img src="bonus_add_files.png" width=30% height=30%>

j. You should get a screen like this!

<img src="bonus_final_mr.png" width=30% height=30%>
