<img src="https://coursesandconferences.wellcomeconnectingscience.org/wp-content/themes/wcc_courses_and_conferences/dist/assets/svg/logo.svg" width="200" height="200">

# Genome Visualisation tools - Paraguay 2023 <!-- omit in toc -->

## Table of contents <!-- omit in toc -->
- [Introduction](#introduction)
  - [Genomic visualisation](#genomic-visualization)
  - [Artemis](#artemis)
- [Practical Exercise](#practical-exercise)


## Module Overview and Aims
The aim of this Module is for you to become familiar with the basic functions of Artemis using a series of worked examples. These examples are designed to take you through the most immediately useful functions. However, there will be time, and encouragement, for you to explore other menus; features of Artemis that are not described in the exercises in this manual, but which may be of particular interest to some users.

# [Introduction](#introduction)

## [Genomic visualisation](#genomic-visualization)

With the advent of Next Generation Sequencing technologies, the amount of genomic data generated has notably increased. Extracting biologically meaningful results from genomic data and communicating this information remains challenging. Visualisation is a key tool to make sense of the trillions of rows of data generated every day. Genome visualisation, e.g. DNA sequence assemblies and annotations and read mappings, is essential for data integration and exploration, interpretation of results and enables scientific discovery and hypothesis generation by providing new insights that may not otherwise be obvious. Additionally, genome visualisation is a valuable aid in expanding and communicating knowledge to both specialized and broad audiences.

Over the last two decades, hundreds of visualisation tools for genomic data have been published. The large number of tools are an indicator for the broad application of genomic data and a sign that visualisation of genomic data is a complex problem and active research domain. Some examples include Tablet, Integrative Genomics Viewer (IGV), GView, CGView, BRIG and Artemis. In this course we will be using Artemis.

## [Artemis](#artemis)

Artemis is a DNA viewer and annotation tool, free to download and use, written by Kim Rutherford from the Sanger Institute (Rutherford et al., 2000). The program allows the user to view a range of files, from simple sequence files (e.g. fasta format) to EMBL/Genbank entries, as well as the results of sequence analyses, in a highly interactive and intuitive graphical format. Artemis is routinely used by the Pathogen Genomics group for annotation and analysis of both prokaryotic and eukaryotic genomes, and can also be used to visualize mapped data from next generation sequencing. Several types/sets of information can be viewed simultaneously within different contexts. For example, Artemis gives you the two views of the same genome region, so you can zoom in to inspect detailed DNA sequence motifs, and also zoom out to view local gene architecture (e.g. operons), or even an entire chromosome or genome, all within one screen. It is also possible to perform analyses within Artemis and save the output for future reference.



